### PR TITLE
[Feature] robotsProps (An Update for robots meta tags)

### PR DESCRIPTION
**Pull request by rodzy**
## Description of Change(s):

Update for the new `robots` meta tags values defined by [Google Search Central - Control Crawling and indexing](https://developers.google.com/search/reference/robots_meta_tag?hl=en-GB#directives).

Instead of using `additionalMetaTags`, that duplicates the meta tag and overrides it, I introduced the new prop `robotsProps` to the `NextSEO` component, this solves the complex integration of new meta values for crawling introduced by Google.

An example here:

```jsx
const Page = () => (
  <>
    <NextSeo
      robotsProps={{
        nosnippet: true,
        notranslate: true,
        noimageindex: true,
        noarchive: true,
        maxSnippet: -1,
        maxImagePreview: 'none',
        maxVideoPreview: -1,
      }}
    />
    <p>Additional robots props in Next-SEO!!</p>
  </>
);

export default Page;
```

### What's new

- `robotsProps` Prop.
- Fully typed.
- `/robots` page to test the additional props.
- Unit test cases for `robotsProps` for `index,follow` / `noindex,nofollow`.
- Integration tests for the new `/robots` page and its `<head>`.
- Documentation section.

---

Some things to help get a PR reviewed and merged faster:

- Have you updated the documentation to go with your changes? **Yes, `#robotsProps`**
- Have you written or updated unit tests? **Yes, I added tests for new meta cases**
- Have you written an integration test in the test app supplied? **Yes, added an integration test for the `robots.tsx` page**

### CODE REVIEW

**Comment by rodzy** [comment]
This solves (#521), @garmeeh let me know what you think about this change, thanks as always 😄

File: src/meta/buildTags.tsx [hunk 1]
```diff
@@ -36,6 +36,28 @@ const buildTags = (config: BuildTagsParams) => {
...
+  let robotsParams = '';
+  if (config.additionalRobotsProps) {
+    const nosnippet = config.additionalRobotsProps.nosnippet;
+    const maxSnippet = config.additionalRobotsProps.maxSnippet;
+    const maxImagePreview = config.additionalRobotsProps.maxImagePreview;
+    const maxVideoPreview = config.additionalRobotsProps.maxVideoPreview;
+    const noarchive = config.additionalRobotsProps.noarchive;
+    const unavailableAfter = config.additionalRobotsProps.unavailableAfter;
+    const noimageindex = config.additionalRobotsProps.noimageindex;
+    const notranslate = config.additionalRobotsProps.notranslate;
```
Comment (garmeeh) [inline src/meta/buildTags.tsx:48]: We could change this to use destructing here instead? 

```
const { nosnippet, maxSnippet, etc } = config.additionalRobotsProps;
```

**Review by garmeeh — CHANGES_REQUESTED** [review]
Excellent work @rodzy 🙌 

Apologies for the delay on getting around to reviewing. 

I have one minor tweak if you wouldn't mind making it. 

Also seeing how you have approached this has given me an idea on how I can close another issue related to not rendering the robots tag by default. One small change that would enable that would be changing `additionalRobotsProps` to `robotsProps` if you had the time?

File: src/meta/buildTags.tsx (hunk 1 above)
Comment (rodzy) [reply src/meta/buildTags.tsx:48]: Thanks for the feedback, for sure I'll be on that this weekend

**Comment by rodzy** [comment]
@garmeeh Changes were quickly applied, I'll be happy to keep helping with future issues 👍

**Review by garmeeh — APPROVED** [review]
Thank you @rodzy 💪

**Comment by garmeeh** [comment]
@rodzy If you would like to contribute some more, I would be more than happy to outline the changes I was thinking of making and maybe you would be interested in making them?

**Comment by rodzy** [comment]
@garmeeh For sure I'm interested, let me know the details and I'll get to it when I got time 👍

**Pull request merged**